crates/bevy_ecs/src/lib.rs

```diff
@@ -1262,6 +1262,37 @@ mod tests {
         assert_eq!(world.resource::<A>().0, 1);
     }
 
+    #[test]
+    fn non_send_resource_scope() {
+        let mut world = World::default();
+        world.insert_non_send_resource(A(0));
+        world.resource_scope(|world: &mut World, mut value: Mut<A>| {
+            value.0 += 1;
+            assert!(!world.contains_resource::<A>());
+        });
+        assert_eq!(world.non_send_resource::<A>().0, 1);
+    }
+
+    #[test]
+    #[should_panic]
```

Comment:
This can probably use `#[should_panic(expected = "fill this in with some error text")]` here.